### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by EmiLoKiJu**
Project requirements:

Create class Person with the following:

- Instance vars: @id, @name, and @age. ✔️ 
- Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true. ✔️ 
- Getters for @id, @name, and @age. ✔️ 
- Setters for @name and @age. ✔️ 
- Private method of_age? that returns true if @age is greater or equal to 18 and false otherwise. ✔️ 
- Public method can_use_services? that returns true if person is of age or if they have permission from parents.✔️ 

Create class Student with the following:

- Inherits from Person. ✔️ 
- Constructor extends parent's constructor by adding @classroom and a parameter for it. ✔️ 
- Method play_hooky that returns "¯\(ツ)/¯". ✔️ 

Create class Teacher with the following:

- Inherits from Person. ✔️ 
- Constructor extends parent's constructor by adding @specialization and a parameter for it. ✔️ 
- Override can_use_services? so it always returns true. ✔️ 

Each class should be saved in a separate file. ✔️

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+class Person
+  def initialize(name = 'Unknown', age = nil, parent_permission: true)
+    @id = SecureRandom.uuid
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_reader :id, :name, :age
```
Comment (brytebee) [inline person.rb:11]: - [ ] **Optional:**
- Conventionally, the `attr` is defined at the top of the class before the initialize method.

Comment (brytebee) [inline person.rb:11]: Kindly make the `id` a read-only `attr`. In a real-world application, IDs are used as references and should not be opened to be modified by the user.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @EmiLoKiJu 👋,

##

### Highlights💡
- Great use of the Ruby style guide✔️
- Great practice with Ruby classes and methods.✔️
- Good commit message.✔️ 
- Good PR Summary.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (EmiLoKiJu) [reply person.rb:11]: @brytebee I don't understand what you mean by making the id a read-only attr. It doesn't have any method to reassign it's value, so it should be read-only, right? 
Am I missing something?

**Comment by ahmetbozaci** [comment]
### Hello @EmiLoKiJu 
### _This is an additional review as a result of the second opinion request._ 


- According to the [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md#:~:text=Getters%20for%20%40id%2C%20%40name,name%20and%20%40age.) getters for ıd, name, age and setters for name and age. It should be like that
image
- You already make the `ıd` read-only.

- [ ] OPTIONAL- Name and age don't define the setters method, and the code reviewer didn't mention it. I recommend adding it. 


___

**_As this was the only reason your project was not approved it will be approved as soon as possible._**

 _Please note that since we will approve it manually so this could **take a few hours**_


### Cheers and Happy coding! 💻 👏
____
[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Pull request merged**